Makefile

```
@@ -151,3 +145,81 @@ docs:  ## Build documentation
 .PHONY: docs-auto
 docs-auto:  ## Build and host docs with sphinx-autobuild
 	uv run --extra docs sphinx-autobuild -b html --open-browser --port=8080 --watch $(PROJECT) -W docs docs/_build
+
+---------------- : ## ----------------
+
+.PHONY: package
+package: package-pip  ## Build all packages
+
+.PHONY: package-pip
+package-pip:  ##- Build packages for pip (sdist, wheel)
+	uv run pyproject-build --installer uv .
+
+.PHONY: package-snap
+package-snap: snap/snapcraft.yaml  ##- Build snap package
+ifeq ($(shell which snapcraft),)
+	sudo snap install --classic snapcraft
+endif
+	snapcraft pack
+
+.PHONY: publish
+publish: publish-pypi  ## Publish packages
+
+.PHONY: publish-pypi
+publish-pypi: clean package-pip  ##- Publish Python packages to pypi
+	uv tool run twine check dist/*
+	uv tool run twine upload dist/*
+
+---------------- : ## ----------------
+
+.PHONY: clean
+clean:  ## Clean up the development environment
+	uv tool run pyclean .
+	rm -rf dist/ build/ docs/_build/ *.snap .coverage*
+
+---------------- : ## ----------------
+
+# Below are intermediate targets for setup. They are not included in help as they should
+# not be used independently.
+
+.PHONY: install-uv
+install-uv:
+ifneq ($(shell which uv),)
+else ifneq ($(shell which snap),)
+	sudo snap install --classic astral-uv
+else ifneq ($(shell which brew),)
+	brew install uv
+else ifeq ($(OS),Windows_NT)
+	pwsh -c "irm https://astral.sh/uv/install.ps1 | iex"
+else
+	curl -LsSf https://astral.sh/uv/install.sh | sh
+endif
```

Maybe avoid the network request if `uv` is already installed:

```suggestion
else
	hash >/dev/null 2>&1 || curl -LsSf https://astral.sh/uv/install.sh | sh
endif
```

But maybe not, if the installer script is smart, small and fast and we want to ensure that the user has the latest `uv` version.

---

`ifneq ($(shell which uv),)` already checked if uv was installed so we do not need to check again I guess?